activation/atxdb.go

```diff
@@ -533,3 +546,20 @@ func (db *ActivationDb) IsIdentityActive(edId string, layer types.LayerID) (*typ
 	// lastAtxTargetEpoch = epoch
 	return &nodeId, atx.TargetEpoch(db.LayersPerEpoch) == epoch, atx.Id(), nil
 }
+
+// ValidateSignedAtx extracts public key from message and verifies public key exists in IdStore, this is how we validate
+// ATX signature. If this is the first ATX it is considered valid anyways and ATX syntactic validation will determine ATX validity
+func (db *ActivationDb) ValidateSignedAtx(pubKey signing.PublicKey, signedAtx *types.ActivationTx) error {
+	// this is the first occurrence of this identity, we cannot validate simply by extracting public key
+	// pass it down to Atx handling so that atx can be syntactically verified and identity could be registered.
+	if signedAtx.PrevATXId == *types.EmptyAtxId {
+		return nil
+	}
+
+	pubString := pubKey.String()
+	_, err := db.GetIdentity(pubString)
+	if err != nil { // means there is no such identity
+		return errInvalidSig
```

Comment:
you should return the err that was received from `GetIdentity` (return it as-is or embedded in a new error)